### PR TITLE
chore: upgrade to Spring Boot 3.1

### PR DESCRIPTION
**Pull request by jnizet**
Note that since Keycloak doesn't seem to support the latest Spring Boot yet, I've switched to the standard Spring Boot oauth2-resource-server, which makes everything easier.

The only downside is that basic-auth is not supported anymore.

### CODE REVIEW

File: backend/src/main/java/fr/inra/urgi/rarebasket/config/SecurityConfig.java [hunk 1]
```diff
@@ -1,83 +1,46 @@
...
+public class SecurityConfig implements WebMvcConfigurer {
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http.csrf(configurer -> configurer.disable())
+            .sessionManagement(customizer -> customizer.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
+            .authorizeHttpRequests(
+                customizer -> customizer.requestMatchers("/api/baskets/**").permitAll()
+                                        .requestMatchers("/api/**").authenticated()
+                                        .anyRequest().permitAll()
```
Comment (cexbrayat) [inline backend/src/main/java/fr/inra/urgi/rarebasket/config/SecurityConfig.java:30]: This is not exactly the previous behavior, right? Some endpoints were not authenticated (like /api/grcs) and are now?

File: backend/src/main/java/fr/inra/urgi/rarebasket/web/user/UserController.java [hunk 2]
```diff
@@ -65,7 +65,10 @@ public UserController(CurrentUser currentUser,
     public UserDTO getCurrent() {
         return currentUser.getId().flatMap(userDao::findById)
                           .map(UserDTO::new)
-                          .orElseThrow(NotFoundException::new);
+                          .orElseThrow(() -> {
+                              System.out.println("Ooops");
```
Comment (cexbrayat) [inline backend/src/main/java/fr/inra/urgi/rarebasket/web/user/UserController.java:69]: Leftover println?

File: backend/src/test/java/fr/inra/urgi/rarebasket/config/TestSecurityConfig.java [hunk 3]
```diff
@@ -9,14 +9,10 @@
...
- * This is necessary because Keycloak Spring support does not work in tests. Yay!
- *
- * (see https://issues.redhat.com/browse/KEYCLOAK-6163, https://issues.redhat.com/browse/KEYCLOAK-4814)
- *
  * @author JB Nizet
  */
-@Configuration
-@Profile("test")
+//@Configuration
+//@Profile("test")
```
Comment (cexbrayat) [inline backend/src/test/java/fr/inra/urgi/rarebasket/config/TestSecurityConfig.java:15]: Comments can be removed ?

**Review by cexbrayat — CHANGES_REQUESTED** [review]

File: backend/src/main/java/fr/inra/urgi/rarebasket/config/SecurityConfig.java (hunk 1 above)
Comment (jnizet) [reply backend/src/main/java/fr/inra/urgi/rarebasket/config/SecurityConfig.java:30]: Indeed. But I think it was a mistake (bug) in the previous version.

**Pull request merged**